pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -921,6 +923,16 @@ cdef class MultiIndexHashTable(HashTable):
                     "hash collision\nlocs:\n{}\n"
                     "result:\n{}\nmi:\n{}".format(alocs, result, mi))
 
+    def _check_for_collision(self, Py_ssize_t loc, object label):
+        # validate that the loc maps to the actual value
+        # version of _check_for_collisions above for single label (tuple)
+
+        result = self.mi[loc]
+        if not array_equivalent(result, label):
```

Comment:
actually I think you need to use ``isnull``; both ``isnan`` and ``math.isnan`` are not complete (IOW they won't for example handle ``NaT``)